corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -0,0 +1,136 @@
+hqDefine("geospatial/js/geospatial_map", [
+    "jquery",
+    "hqwebapp/js/initial_page_data",
+], function (
+    $,
+    initialPageData,
+) {
+    $(function () {
+        var loadMapBox = function (centerCoordinates) {
+            'use strict';
+
+            var self = {};
+            var markers = [];
+            mapboxgl.accessToken = initialPageData.get('mapbox_access_token');
+
+            if (!centerCoordinates) {
+                centerCoordinates = [-91.874, 42.76]; // should be domain specific
+            }
+
+            const map = new mapboxgl.Map({
+                container: 'geospatial-map', // container ID
+                style: 'mapbox://styles/mapbox/streets-v12', // style URL
+                center: centerCoordinates, // starting position [lng, lat]
+                zoom: 12,
+                attribution: '© <a href="https://www.mapbox.com/about/maps/">Mapbox</a> ©' +
+                             ' <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+            });
+
+            const draw = new MapboxDraw({
+                // API: https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md
+                displayControlsDefault: false,
+                boxSelect: true, // enables box selection
+                controls: {
+                    polygon: true,
+                    trash: true
+                },
+            });
+            map.addControl(draw);
+
+            map.on('draw.create', updateArea);
+            map.on('draw.delete', updateArea);
+            map.on('draw.update', updateArea);
+
+            function getCoordinates(event) {
+                return event.lngLat;
+            };
+
+            function updateArea(e) {
+                const data = draw.getAll();
+                const area = turf.area(data);
+                // Restrict the area to 2 decimal points.
+                const rounded_area = Math.round(area * 100) / 100;
+            }
+
+            // We should consider refactoring and splitting the below out to a new JS file
```

Comment:
Agreed - it doesn't need to block this PR, but it could be quite useful to have relatively independent mapping util functions that could be used here but also in other maps in HQ (thinking specifically of web apps).

Reply:
Yeah. We can see how to move it into the existing `map_utils.js` file